controllers/controllerhelpers/chatLogger.go

```diff
@@ -0,0 +1,124 @@
+package controllerhelpers
+
+import (
+	"fmt"
+	"github.com/TF2Stadium/Helen/config"
+	"github.com/TF2Stadium/Helen/helpers"
+	"github.com/TF2Stadium/Helen/models"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+)
+
+var mapLock = &sync.Mutex{}
+var roomLogChannel = make(map[uint](chan string))
+
+var globalLog *os.File
+var globalLogLock = &sync.Mutex{}
+var globalLogTicker = time.Tick(time.Hour * 24)
+
+func StartGlobalLogger() {
+	go globalLogFileUpdater()
+}
+
+func globalLogFileUpdater() {
+	init := true
+	var now time.Time
+	for {
+		if !init {
+			now = <-globalLogTicker
+		} else {
+			now = time.Now()
+			roomLogChannel[0] = make(chan string, 18)
+		}
+		globalLogLock.Lock()
```

Comment:
I can't see a blocking function in here. If there isn't one, this will keep creating files all the time.